# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit/CacheIR.cpp
# Commit: 704d04e6bc76
# Full Hash: 704d04e6bc76e06107a2a8856cef2363f70f73fb
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2020-06-24 16:24:33
# Description:
#   Bug 1646302 - Only create a template object if prototype is an object in getTemplateObjectForScripted. r=evilpie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D80595
# ==============================================================================

diff -r e29a263455f8 -r 704d04e6bc76 js/src/jit/CacheIR.cpp
--- a/js/src/jit/CacheIR.cpp	Wed Jun 24 09:13:28 2020 +0000
+++ b/js/src/jit/CacheIR.cpp	Tue Jun 23 10:11:10 2020 +0000
@@ -6347,7 +6347,11 @@
     return true;
   }
 
-  if (protov.isObject()) {
+  if (!protov.isObject()) {
+    return true;
+  }
+
+  {
     AutoRealm ar(cx_, calleeFunc);
     TaggedProto proto(&protov.toObject());
     ObjectGroup* group = ObjectGroup::defaultNewGroup(cx_, &PlainObject::class_,
